backend_new/src/controllers/listing.controller.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  ClassSerializerInterceptor,
+  Controller,
+  Get,
+  Headers,
+  Param,
+  ParseUUIDPipe,
+  Query,
+  UseInterceptors,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import {
+  ApiExtraModels,
+  ApiOperation,
+  ApiTags,
+  ApiOkResponse,
+} from '@nestjs/swagger';
+import { ListingService } from '../services/listing.service';
+import { defaultValidationPipeOptions } from '../utilities/default-validation-pipe-options';
+import { ListingsQueryParams } from '../dtos/listings/listings-query-params.dto';
+import { LanguagesEnum } from '@prisma/client';
+import { ListingsRetrieveParams } from '../dtos/listings/listings-retrieve-params.dto';
+import { PaginationAllowsAllQueryParams } from '../dtos/shared/pagination.dto';
+import { ListingFilterParams } from '../dtos/listings/listings-filter-params.dto';
+import { PaginatedListingDto } from '../dtos/listings/paginated-listing.dto';
+import ListingGet from '../dtos/listings/listing-get.dto';
+
+@Controller('listings')
+@ApiTags('listings')
+@ApiExtraModels(
+  ListingsQueryParams,
+  ListingFilterParams,
+  ListingsRetrieveParams,
+  PaginationAllowsAllQueryParams,
+)
+export class ListingController {
+  constructor(private readonly listingService: ListingService) {}
+
+  @Get()
+  @ApiOperation({
+    summary: 'Get a paginated set of listings',
+    operationId: 'list',
+  })
+  @UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+  @UseInterceptors(ClassSerializerInterceptor)
+  @ApiOkResponse({ type: PaginatedListingDto })
+  public async getPaginatedSet(@Query() queryParams: ListingsQueryParams) {
+    return await this.listingService.list(queryParams);
+  }
+
+  @Get(`:id`)
+  @ApiOperation({ summary: 'Get listing by id', operationId: 'retrieve' })
+  @UseInterceptors(ClassSerializerInterceptor)
+  @UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+  @ApiOkResponse({ type: ListingGet })
+  async retrieve(
+    @Headers('language') language: LanguagesEnum,
+    @Param('id', new ParseUUIDPipe({ version: '4' })) listingId: string,
+    @Query() queryParams: ListingsRetrieveParams,
+  ) {
+    if (listingId === undefined || listingId === 'undefined') {
```

Comment:
Just curious why we need to check for both

Reply:
tbh I'm unsure, that was in place in our typeorm version of this controller so I kept it in

good call out though because the `ParseUUIDPipe` should catch the `'undefined'` case